src/app/src/androidTest/java/com/example/bookmark/SignupActivityTest.java

```diff
@@ -0,0 +1,68 @@
+package com.example.bookmark;
+
+import android.app.Activity;
+import android.view.View;
+
+import androidx.test.platform.app.InstrumentationRegistry;
+import androidx.test.rule.ActivityTestRule;
+
+import com.robotium.solo.Solo;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+
+/**
+ * Perform intent testing on the SignupActivity
+ * <p>
+ * Outstanding Issues/TODOs
+ * Test Signing Up (how to make sure that user exists?)
+ *
+ * @author Konrad Staniszewski.
+ */
+public class SignupActivityTest {
+    private Solo solo;
+    @Rule
+    public ActivityTestRule<SignupActivity> rule =
+        new ActivityTestRule<>(SignupActivity.class, true, true);
+
+    /**
+     * Runs before all tests and create the solo instance.
+     *
+     * @throws Exception
+     */
+    @Before
+    public void setUp() throws Exception {
+        solo = new Solo(InstrumentationRegistry.getInstrumentation(), rule.getActivity());
+    }
+
+    /**
+     * Gets activity
+     *
+     * @throws Exception
+     */
+    @Test
+    public void start() throws Exception {
+        Activity activity = rule.getActivity();
+    }
+
+    /**
+     * Test going back to login
+     */
+    @Test
+    public void backToLogin() {
+    }
```

Comment:
I assume this is something you have yet to implement?

Reply:
It needs the MockFirebaseProvider, but yes.